### ISSUE TITLE
Circular reference when outputting JSON with generic type array

### ISSUE THREAD

**Issue by williamkbentley** [TypeStrong/typedoc#1131]
- [x] I have checked [issues with bug label](https://github.com/TypeStrong/typedoc/labels/bug) and found no duplicates

## Expected Behavior

JSON produced

## Actual Behavior

TypeError: Converting circular structure to JSON

## Steps to reproduce the bug

Consider a project with a single file:
```ts
  export class Foo {
  }

  export class FooList extends Array<Foo> {  }
```

This works with `--out`, but produces the above circular structure error with option `--json`. I have isolated the issue down to the `FooList` line.
Works in 0.15.0.

## Environment
  - Typedoc version: 0.15.1
  - Node.js version: 10.15.0
  - OS: Windows

**Comment by Gerrit0** [comment]
Well, at least it was easy to track down.... I should have caught this. Introduced by 0a61a00a69566d7935d79bafbdb5ef5734932a7d

**Comment by Gerrit0** [comment]
But apparently this reveals another bug... Which I knew existed, but now feel like I should fix for this.

**Comment by Gerrit0** [comment]
Fix published in v0.15.2
